pkg/gadgets/trace/exec/tracer/tracer_test.go

```diff
@@ -287,6 +289,12 @@ func TestExecTracer(t *testing.T) {
 				// normalize
 				event.Timestamp = 0
 
+				// When tests are run with "go test", the binary is named tracer.test
+				// When tests are run from Goland, the binary is named ___TestExecTrac
+				if event.Pcomm == "___TestExecTrac" {
+					event.Pcomm = "tracer.test"
+				}
+
```

Comment:
What about this? It should work in all the cases:

```diff
diff --git pkg/gadgets/trace/exec/tracer/tracer_test.go pkg/gadgets/trace/exec/tracer/tracer_test.go
index 70e98bef9..98c9ae0b2 100644
--- pkg/gadgets/trace/exec/tracer/tracer_test.go
+++ pkg/gadgets/trace/exec/tracer/tracer_test.go
@@ -21,6 +21,7 @@ import (
        "fmt"
        "os"
        "os/exec"
+       "path"
        "sort"
        "testing"
        "time"
@@ -72,6 +73,10 @@ func TestExecTracer(t *testing.T) {
        cwd, err := os.Getwd()
        require.Nil(t, err, "Failed to get current working directory: %s", err)
 
+       executable, err := os.Executable()
+       require.Nil(t, err, "Failed to get executable path: %s", err)
+       pcomm := path.Base(executable)
+
        type testDefinition struct {
                shouldSkip      func(t *testing.T)
                getTracerConfig func(info *utilstest.RunnerInfo) *tracer.Config
@@ -102,7 +107,7 @@ func TestExecTracer(t *testing.T) {
                                        WithMountNsID: eventtypes.WithMountNsID{MountNsID: info.MountNsID},
                                        Retval:        0,
                                        Comm:          "cat",
-                                       Pcomm:         "tracer.test",
+                                       Pcomm:         pcomm,
                                        Args:          []string{"/bin/cat", "/dev/null"},
                                }
                        }),
@@ -136,7 +141,7 @@ func TestExecTracer(t *testing.T) {
                                        WithMountNsID: eventtypes.WithMountNsID{MountNsID: info.MountNsID},
                                        Retval:        0,
                                        Comm:          "cat",
-                                       Pcomm:         "tracer.test",
+                                       Pcomm:         pcomm,
                                        Args:          []string{"/bin/cat", "/dev/null"},
                                }
                        }),
@@ -288,13 +293,6 @@ func TestExecTracer(t *testing.T) {
                        eventCallback := func(event *types.Event) {
                                // normalize
                                event.Timestamp = 0
-
-                               // When tests are run with "go test", the binary is named tracer.test
-                               // When tests are run from Goland, the binary is named ___TestExecTrac
-                               if event.Pcomm == "___TestExecTrac" {
-                                       event.Pcomm = "tracer.test"
-                               }
-
                                events = append(events, *event)
                        }
```

Reply:
It works! Thanks for the idea.